src/gt4py/ir/utils.py

```diff
@@ -30,6 +31,135 @@
 DEFAULT_LAYOUT_ID = "_default_layout_id_"
 
 
+class AxisBoundParser(ast.NodeVisitor):
```

Comment:
This should really be in `gtscript_frontend.py` but because of the unused `AST2IRVisitor` we need it here to avoid a circular dependency.

@egparedes Can I just remove the unused visitor and disabled tests, or is there logic that we haven't carried over yet?

Reply:
I had missed this comment.... IIRC, the mentioned `AST2IRVisitor` and most of the tools in `ir/utils.py` were just used as helpers to create IR data for some code generation tests. If those tests are not longer used, and since those helpers are not properly documented and unlikely to be needed again, I think it makes sense to remove them, together with other commented out parts in the frontend/ir modules.

If the pieces of dead code to be removed are too large to keep the focus of this PR, another possibility could be to open a new small PR just removing dead code, and then finish this one on top of it.

Reply:
I like that approach. I'll do that today.

Reply:
See #244.